autovalet_goal_generation/src/autovalet_goal_generation/goal_generator.py

```diff
@@ -44,30 +46,65 @@ def generate_goal_from_egoline(self, ego_line, egoline_frame):
         '''
 
         # ref: https://stackoverflow.com/questions/2298390/fitting-a-line-in-3d/2333251#2333251
+        ego_line         = ego_line[ego_line[:, 2].argsort()]
+        #target_point     = ego_line[int(ego_line.shape[0]/4)]
         egoline_midpoint = ego_line.mean(axis=0)
         _, _, Vt         = np.linalg.svd(ego_line - egoline_midpoint)
         line_direction   = Vt[0]    # The principal direction of the distribution. Line direction here
-        
+
         # Correct sign to ensure that the direction is always away from the robot
         # The fix is to ensure the z-direction is always positive which means direction away from the robot
         line_direction = np.sign(line_direction[-1])*line_direction
 
+        # generate target pos
+        dist_to_costmap  = egoline_midpoint[2] - 7.0
```

Comment:
can you maybe read this value from the parameter server? should be `/move_base/global_costmap/height` or width im not sure which. this way allows us to change yaml file without changing code

Reply:
done